atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/backup/CassandraRepairHelper.java

```diff
@@ -0,0 +1,130 @@
+import java.util.stream.Collectors;
+
+public class CassandraRepairHelper {
+    private static final Set<TableReference> TABLES_TO_REPAIR =
+            Sets.union(ImmutableSet.of(AtlasDbConstants.COORDINATION_TABLE), TargetedSweepTables.REPAIR_ON_RESTORE);
+
+    private final MetricsManager metricsManager;
+    private final Function<Namespace, CassandraKeyValueServiceConfig> keyValueServiceConfigFactory;
+    private final Function<Namespace, KeyValueService> keyValueServiceFactory;
+
+    public CassandraRepairHelper(
+            MetricsManager metricsManager,
+            Function<Namespace, CassandraKeyValueServiceConfig> keyValueServiceConfigFactory,
+            Function<Namespace, KeyValueService> keyValueServiceFactory) {
+        this.metricsManager = metricsManager;
+        this.keyValueServiceConfigFactory = keyValueServiceConfigFactory;
+        this.keyValueServiceFactory = keyValueServiceFactory;
+    }
+
+    public void repairInternalTables(
+            Namespace namespace, Consumer<Map<InetSocketAddress, Set<LightweightOppTokenRange>>> repairTable) {
+        KeyValueService kvs = keyValueServiceFactory.apply(namespace);
+        kvs.getAllTableNames().stream()
+                .filter(TABLES_TO_REPAIR::contains)
+                .map(TableReference::getTableName)
+                .map(tableName -> getRangesToRepair(namespace, tableName))
+                .forEach(repairTable);
+    }
+
+    // VisibleForTesting
+    public Map<InetSocketAddress, Set<LightweightOppTokenRange>> getRangesToRepair(
+            Namespace namespace, String tableName) {
+        Map<InetSocketAddress, Set<TokenRange>> tokenRanges = getTokenRangesToRepair(namespace, tableName);
+        return KeyedStream.stream(tokenRanges).map(this::makeLightweight).collectToMap();
+    }
+
+    private Map<InetSocketAddress, Set<TokenRange>> getTokenRangesToRepair(Namespace namespace, String tableName) {
+        CassandraKeyValueServiceConfig config = keyValueServiceConfigFactory.apply(namespace);
+        String cassandraTableName = getCassandraTableName(namespace, tableName);
+        return CqlCluster.create(config).getTokenRanges(cassandraTableName);
+    }
+
+    private String getCassandraTableName(Namespace namespace, String tableName) {
+        TableReference tableRef = TableReference.create(toKvNamespace(namespace), tableName);
+        return AbstractKeyValueService.internalTableName(tableRef);
+    }
+
+    private com.palantir.atlasdb.keyvalue.api.Namespace toKvNamespace(Namespace namespace) {
+        return com.palantir.atlasdb.keyvalue.api.Namespace.create(namespace.get());
+    }
+
+    private Set<LightweightOppTokenRange> makeLightweight(Set<TokenRange> tokenRanges) {
+        return tokenRanges.stream()
+                .map(this::makeLightweight)
+                .flatMap(Collection::stream)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<LightweightOppTokenRange> makeLightweight(TokenRange tokenRange) {
```

Comment:
I guess this could return a stream to make it nicer. It is probably correct for us to serialise this, especially since we already may have had a bug here?!